inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/experimental_detectron_roi_feature_extractor_ref.cl

```diff
@@ -0,0 +1,131 @@
+    INPUT0_TYPE roi_end_h = current_roi_ptr[3] * spatial_scale - offset;
+
+    INPUT0_TYPE roi_width = max(roi_end_w - roi_start_w, TO_INPUT0_TYPE(1.));
+    INPUT0_TYPE roi_height = max(roi_end_h - roi_start_h, TO_INPUT0_TYPE(1.));
+
+    INPUT0_TYPE bin_width = roi_width / TO_INPUT0_TYPE(POOLED_WIDTH);
+    INPUT0_TYPE bin_height = roi_height / TO_INPUT0_TYPE(POOLED_HEIGHT);
+
+    const uint roi_bin_grid_w = (SAMPLING_RATIO > 0) ? SAMPLING_RATIO : (uint)ceil(roi_width / POOLED_WIDTH);
+    const uint roi_bin_grid_h = (SAMPLING_RATIO > 0) ? SAMPLING_RATIO : (uint)ceil(roi_height / POOLED_HEIGHT);
+
+    const uint level_h = LEVEL_SIZES[3 * level];
+    const uint level_w = LEVEL_SIZES[3 * level + 1];
+    const uint level_offset = LEVEL_SIZES[3 * level + 2];
+
+    INPUT1_TYPE output_val = 0.0;
+    const __global INPUT1_TYPE* data = current_level_ptr + level_offset + c * level_h * level_w;
+
+    INPUT0_TYPE current_bin_start_h = roi_start_h + y * bin_height;
+    INPUT0_TYPE current_bin_start_w = roi_start_w + x * bin_width;
+    for (int iy = 0; iy < roi_bin_grid_h; iy++) {
+        INPUT0_TYPE yy = current_bin_start_h + TO_INPUT0_TYPE(iy + 0.5f) * bin_height / TO_INPUT0_TYPE(roi_bin_grid_h);
+        if (yy < -1.0 || yy > level_h) {
+            continue;
+        }
+        if (yy <= 0) {
+            yy = 0.0f;
+        }
+        int y_low = (int)floor(yy);
+        int y_high = 0;
+
+        if (y_low >= level_h - 1) {
+            y_high = y_low = level_h - 1;
+            yy = TO_INPUT0_TYPE(y_low);
+        } else {
+            y_high = y_low + 1;
+        }
+
+        INPUT0_TYPE ly = yy - y_low;
+        INPUT0_TYPE hy = TO_INPUT0_TYPE(1.0f) - ly;
+
+        for (int ix = 0; ix < roi_bin_grid_w; ix++) {
+            INPUT0_TYPE xx = current_bin_start_w + TO_INPUT0_TYPE(ix + 0.5f) * bin_width / TO_INPUT0_TYPE(roi_bin_grid_w);
+            if (xx < -1.0 || xx > level_w) {
+                continue;
+            }
+            if (xx <= 0) {
+                xx = 0.0f;
+            }
+            int x_low = (int)floor(xx);
+            int x_high = 0;
+
+            if (x_low >= level_w - 1) {
+                x_high = x_low = level_w - 1;
+                xx = TO_INPUT0_TYPE(x_low);
+            } else {
+                x_high = x_low + 1;
+            }
+
+            INPUT0_TYPE lx = xx - x_low;
+            INPUT0_TYPE hx = TO_INPUT0_TYPE(1.0f) - lx;
+
+            INPUT0_TYPE w1 = hy * hx;
+            INPUT0_TYPE w2 = hy * lx;
+            INPUT0_TYPE w3 = ly * hx;
+            INPUT0_TYPE w4 = ly * lx;
+
+            output_val += w1 * data[y_low * level_w + x_low] +
+                          w2 * data[y_low * level_w + x_high] +
+                          w3 * data[y_high * level_w + x_low] +
+                          w4 * data[y_high * level_w + x_high];
+        }
+    }
+    output_val /= TO_INPUT0_TYPE(roi_bin_grid_h * roi_bin_grid_w);
+    dst_data[i] = output_val;
```

Comment:
Take a look please, in your code:
 `dst_data` has type `OUTPUT_TYPE`
 `output_val` has type `INPUT1_TYPE`
And on line 129 you use cast to `INPUT0_TYPE` before saving the result into the `output_val`.
So you use 3 different data types for calculating and saving the output.

Reply:
Type of output_val was changed on INPUT0_TYPE